crates/kilt-dip-support/src/lib.rs

```diff
@@ -0,0 +1,92 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2023 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+// TODO: Crate documentation
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ::did::DidSignature;
+use pallet_dip_consumer::traits::IdentityProofVerifier;
+use sp_std::marker::PhantomData;
+
+pub mod did;
+pub mod merkle;
+pub mod traits;
+
+/// A type that chains a Merkle proof verification with a DID signature
```

Comment:
The logic here is a bit more convoluted. 
Inside the code, we do:

```rust
	type IdentityDetails = <DidSignatureVerifier as IdentityProofVerifier<Call, Subject>>::IdentityDetails;
	type Submitter = <MerkleProofVerifier as IdentityProofVerifier<Call, Subject>>::Submitter;
	type VerificationResult = <MerkleProofVerifier as IdentityProofVerifier<Call, Subject>>::VerificationResult;
```

So this case is rather specific to the process of: 1. verifying the Merkle proof, 2. use the additional details + the result of the Merkle proof to verify the DID signature. It would be hard to chain them in a generic way, since the DID signature  verifier requires the result of the Merkle verification + the DID signature and block number. I agree probably the traits are not really a good fit here. I'll remove them.

Reply:
Double-thinking this, I think it's ok to leave traits here. Yes, it's not generic, but for instance the DID verifier trait might or might not include the check on the call type. Having a generic `Chained<A, B>` type might make sense, but the provided default would still be the one provided here. Hence, I think, at least for now, I will not integrate this comment, although time and other projects will tell whether we need a more generic solution here.
For now, the only reason to have traits here, is that the individual components, i.e., the Merkle verifier and the DID signature verifier, can be swapped with other types that implement the same trait. For instance, for the DID verifier, it could perform the additional call filtering or not, depending on the use case.